internal/plugin/server.go

```diff
@@ -354,8 +354,8 @@ func (plugin *NvidiaDevicePlugin) getAllocateResponse(requestIds []string) (*plu
 	if plugin.config.Sharing.SharingStrategy() == spec.SharingStrategyMPS {
 		plugin.updateResponseForMPS(response)
 	}
-	if *plugin.config.Flags.Plugin.PassDeviceSpecs {
-		response.Devices = append(response.Devices, plugin.apiDeviceSpecs(*plugin.config.Flags.NvidiaDriverRoot, requestIds)...)
+	if *plugin.config.Flags.Plugin.PassDeviceSpecs && !plugin.deviceListStrategies.IsNonCDIEnabled() {
```

Comment:
This double-negative is a bit difficult to read. Would we be able to update this to `*plugin.config.Flags.Plugin.PassDeviceSpecs && plugin.deviceListStrategies.IsCDIEnabled()`?

Also, I am not quite sure I am following this check. Why does it matter that CDI is enabled when `--pass-device-specs` is set?

Reply:
The reason for the check is that if *only* CDI is used, then we don't want to inject the device specs in this way since CDI is used to inject the device nodes.

Let me see if I can come up with something that's clearer.